### ISSUE TITLE
Cannot use non-login shell by default

### ISSUE THREAD

**Issue by mxfm** [tmux/tmux#1937]
By default tmux opens login shell which is documented in man. This is not desirable in many cases because running login shell invokes '.bash_profile' which are intended to run only at login, while tmux can be run started and exited at any time after login. This means that 1) login scripts are executed after login and 2) they are executed several times.

This issue can be resolved by option 'default shell' in tmux conf as ''set-option default-shell /bin/bash', however this does not work because during reading config file there seems to be no active sessions ('/home/user/.tmux.conf:1: no current session').

So, this is issue cannot be resolved by tmux, only by tweaking '.bash_profile', but this is not a proper solution. 

* tmux version (`tmux -V`). tmux 2.9a
* tmux config: 1-line 'set-option default-shell /bin/bash'
* Platform (`uname -sp`). Archlinux
* $TERM inside and outside of tmux (`echo $TERM`). 'screen' and 'xterm-256color' (outside)
* Logs from tmux (`tmux kill-server; tmux -vv new`). '/home/user/.tmux.conf:1: no current session'

**Comment by nicm** [comment]
Set default-command.

**Comment by mxfm** [comment]
in .tmux.conf:
"set-option default-command /bin/bash"
After running:
/home/user/.tmux.conf:1: no current session

**Comment by nicm** [comment]
set -g

**Comment by nicm** [comment]
You probably want "exec /bin/bash" also.

**Comment by mxfm** [comment]
Ok, it seems that "set -g default-command "exec /bin/bash"" in '.tmux.conf' fixes the issue.

**Comment by krader1961** [comment]
@mxfm I am curious what you think "run only at login" means with respect to config file *~/.bash_profile*. Most virtual terminals such as xterm, iTerm, etc. run the shell associated with your profile as a "login" shell. Yet you have already logged into your system.

This is mostly a rhetorical question. But I ask it because you don't seem to understand what is meant by "login shell".

**Comment by mxfm** [comment]
@krader1961 The idea behind login shell is to configure user session: to set environment variables and to do some other work, which is preserved across running other shells. This is usually done at login which makes sense (unfortunately the man page of bash tells about the difference only in terms of which files are processed, not *why* does the difference exists). As an exception, X applications often need also some configuration, so grahphical managers can also invoke login shell. But in other circumstances running login shell is not desirable behavior for the reasons I outlined above (does not make sense & can run login scripts several times).

> Most virtual terminals such as xterm, iTerm, etc. run the shell associated with your profile as a "login" shell.

This is not the case of xterm (perhaps in other distros it is compiled to run a login shell?) - I called it in several cases previously directly when I tried to avoid running login scripts - it does not act as a login shell. I cannot say for iTerm. It looks like you were using virtual terminals which run login shell by default and consider this normal behavior.
